api/include/opentelemetry/context/threadlocal_context.h

```diff
@@ -0,0 +1,121 @@
+#pragma once
+
+#include "opentelemetry/context/runtime_context.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace context
+{
+
+// The ThreadLocalContext class is a derived class from RuntimeContext and
+// provides a wrapper for propogating context through cpp thread locally.
+class ThreadLocalContext : public RuntimeContext
```

Comment:
I don't understand the relationship between ThreadLocalContext and RuntimeContext. Part of what's confusing me is the static methods of threadlocal, and that there's no implementation for the member functions of RuntimeContext itself.  Is ThreadLocalContext meant to override methods of RuntimeContext?

Whatever the answer is, some more class-level comments about how they are meant to be used would help.

Reply:
So my understanding is that the RuntimeContext provides the capability for users to implement their own version of threadlocal_context. The threadlocal_context methods are static because shouldn't the context be available globally. Please correct me if any of this doesn't make sense though, one way or another I will add more comments.

Reply:
The threadlocal context is meant to override methods of RuntimeContext

Reply:
If the methods need to be overridden then they should be virtual in RuntimeContext, and ThreadLocalContext should use `override`

That said, `static` member functions cannot be overridden that way.

The fact that there's a RuntimeContext base suggests that it's the thing that users are supposed to interact with (i.e. they should call RuntimeContext::GetCurrent(), but the implementation would be in ThreadLocalContext. Does that match what you're aiming for?

Reply:
I don't think that matches what I was hearing from reiley.

Reply:
But I am probably mistaken, I should check the python implementaion, because I think that's what he wants me to match.